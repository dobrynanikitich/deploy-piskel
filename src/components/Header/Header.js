import React, { Component } from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import classes from './Header.module.scss';

class Header extends Component {
    state = {
        userName: 'SIGN IN',
    }

    logInHandler = () => {
            const { changeState } = this.props;
            const auth2 = window.gapi.auth2.getAuthInstance();
            auth2.signIn().then((googleUser) => {
            const myProfile = googleUser.getBasicProfile();
            const userName = myProfile.getGivenName();
            this.setState({ userName: userName });
            changeState('isSignedIn', true)
        });
    }

    logOutHandler = () => {
        const { changeState } = this.props;
        const auth2 = window.gapi.auth2.getAuthInstance();
        auth2.signOut();
        this.setState({ userName: 'SIGN IN' });
        changeState('isSignedIn', false);
    }

    render() {
        const { isSignedIn } = this.props;
        const { userName } = this.state;
        return (
            <header className={classes.Header}>
                <div className={classes.HeaderName}>PISKEL CLONE</div>
                <div className={classes.BtnsWrapper}>
                    <a href='#/' className={classes.BackToLanding}>
                        BACK
                    </a>
                    <div 
                        className={classes.AuthWrapper}
                        onClick={isSignedIn ? this.logOutHandler : this.logInHandler}
                        data-testid="sign-in-test"
                        >
                        <FontAwesomeIcon icon={['fab', 'google']}/>
                        <div
                            className={classes.AuthSection} 
                            >{userName}
                        </div>
                        {isSignedIn ? <FontAwesomeIcon icon={'sign-out-alt'}/> : ''}
                    </div>
                </div>
            </header>
        )
    }
}

export default Header;