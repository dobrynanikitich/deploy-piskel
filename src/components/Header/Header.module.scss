.Header {
    width: 100%;
    height: 55px;
    background: darkslategray;
    display: flex;
    justify-content: space-between;
    align-items: center;
    box-sizing: border-box;
    border-bottom: 3px solid goldenrod;
    font-size: 25px;
    font-weight: bold;
    padding-left: 15px;
    .HeaderName {
        font-size: 40px;
        color: salmon;
    }
    .BtnsWrapper {
        display: flex;
    }
    .BackToLanding {
        display: flex;
        align-items: center;
        justify-content: center;
        background: goldenrod;
        height: 45px;
        color: white;
        min-width: 150px;
        width: fit-content;
        box-sizing: border-box;
        padding: 15px;
        margin-right: 10px;
        text-decoration: none;
    }
    .AuthWrapper {
        display: flex;
        align-items: center;
        background: white;
        height: 45px;
        color: darkblue;
        min-width: 150px;
        width: fit-content;
        box-sizing: border-box;
        padding: 15px;
        margin-right: 10px;
        &:hover {
            cursor: pointer;
        }
    }
    .AuthSection {
        color: orangered;
        font-size: 15px;
        width: 100%;
        display: flex;
        justify-content: space-evenly;
    }
}