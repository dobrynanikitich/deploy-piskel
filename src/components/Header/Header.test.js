import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import sinon from 'sinon';

import classes from './Header.module.scss';
import Header from './Header';

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

Enzyme.configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const wrapper = shallow(
    <Header 
      isSignedIn={true}
      changeState={() => {}}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly again', () => {
  const wrapper = render(
    <Header 
      isSignedIn={true}
      changeState={() => {}}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('formats application name correctly', () => {
  const wrapper = mount(
    <Header 
      isSignedIn={true}
      changeState={() => {}}
    />
  );

  const text = wrapper.find('div').first().text()

  expect(text).toEqual('PISKEL CLONE');
});

it('back button doesnt calls', () => {
  const spy = sinon.spy();
  const wrapper = mount(
    <a href='#/' className={classes.BackToLanding}>
        BACK
    </a>
  );

  wrapper.simulate('click');

  expect(spy.calledOnce).toBe(false);
});

// it('sign in button clicks correctly', () => {
//   const spy = sinon.spy();
//   const wrapper = mount(
//     <Header 
//       isSignedIn={false}
//       onClick={spy}
//     />
//   );

//   wrapper.find('[data-testid="sign-in-test"]').simulate('click');
//   expect(spy.calledOnce).toBe(true); 
// });
