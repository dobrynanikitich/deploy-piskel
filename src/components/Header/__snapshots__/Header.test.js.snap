// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="Header"
>
  <div
    className="HeaderName"
  >
    PISKEL CLONE
  </div>
  <div
    className="BtnsWrapper"
  >
    <a
      className="BackToLanding"
      href="#/"
    >
      BACK
    </a>
    <div
      className="AuthWrapper"
      data-testid="sign-in-test"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Array [
            "fab",
            "google",
          ]
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <div
        className="AuthSection"
      >
        SIGN IN
      </div>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="sign-out-alt"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </div>
  </div>
</header>
`;

exports[`renders correctly again 1`] = `
<header
  class="Header"
>
  <div
    class="HeaderName"
  >
    PISKEL CLONE
  </div>
  <div
    class="BtnsWrapper"
  >
    <a
      class="BackToLanding"
      href="#/"
    >
      BACK
    </a>
    <div
      class="AuthWrapper"
      data-testid="sign-in-test"
    >
      <div
        class="AuthSection"
      >
        SIGN IN
      </div>
    </div>
  </div>
</header>
`;
